### PR TITLE
feat(private-network): add wait process for private networks on instance

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -857,6 +857,8 @@ github.com/safchain/ethtool v0.0.0-20210803160452-9aa261dae9b1/go.mod h1:Z0q5wiB
 github.com/satori/go.uuid v1.2.0/go.mod h1:dA0hQrYB0VpLJoorglMZABFdXlWrHn1NEOzdhQKdks0=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.9.0.20220308141350-165c40ac312e h1:clWNRWjqkdaqZ2zUCKnzS5fPypivPeuZyKpy0fNe9zU=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.9.0.20220308141350-165c40ac312e/go.mod h1:fCa7OJZ/9DRTnOKmxvT6pn+LPWUptQAmHF/SBJUGEcg=
+github.com/scaleway/scaleway-sdk-go v1.0.0-beta.9.0.20220413124200-87ec17340712 h1:MOkU+rQ1ZIIqYAjCUwd1qaOapaECgic6kZjuT+iK2/Y=
```
Comment (remyleone) [inline go.sum:860]: go mod tidy

File: scaleway/helpers_instance.go [hunk 2]
```diff
@@ -331,10 +331,21 @@ func (ph *privateNICsHandler) flatPrivateNICs() error {
...
 	if oPtr != nil && len(*oPtr) > 0 {
+		idPN := expandID(*oPtr)
 		// check if old private network still exist on instance server
-		if p, ok := ph.privateNICsMap[expandID(*oPtr)]; ok {
+		if p, ok := ph.privateNICsMap[idPN]; ok {
+			_, err := ph.instanceAPI.WaitForPrivateNIC(&instance.WaitForPrivateNICRequest{
+				ServerID:      ph.serverID,
+				PrivateNicID:  expandID(p.ID),
+				Zone:          ph.zone,
+				Timeout:       scw.TimeDurationPtr(d.Timeout(schema.TimeoutCreate)),
```
Comment (remyleone) [inline scaleway/helpers_instance.go:341]: d is not defined and you don't know if it is called from a create. Timeout should be passed from the function Create, Read, Update, Delete.

File: scaleway/resource_instance_private_nic.go [hunk 3]
```diff
@@ -81,7 +81,7 @@ func resourceScalewayInstancePrivateNICRead(ctx context.Context, d *schema.Resou
 		return diag.FromErr(err)
 	}
 
-	res, err := instanceAPI.GetPrivateNIC(&instance.GetPrivateNICRequest{
+	res, err := instanceAPI.WaitForPrivateNIC(&instance.WaitForPrivateNICRequest{
```
Comment (remyleone) [inline scaleway/resource_instance_private_nic.go:84]: Should be replaced by a waitForPrivateNIC using the same pattern that was recently merged that allow for quick retries

File: scaleway/resource_instance_private_nic.go [hunk 4]
```diff
@@ -114,6 +114,15 @@ func resourceScalewayInstancePrivateNICUpdate(ctx context.Context, d *schema.Res
 	}
 
 	if d.HasChanges("private_network_id", "server_id") {
+		_, err := instanceAPI.WaitForPrivateNIC(&instance.WaitForPrivateNICRequest{
```
Comment (remyleone) [inline scaleway/resource_instance_private_nic.go:117]: You should pass the `d.Timeout(schema.TimeoutUpdate)` from here.

File: scaleway/resource_instance_private_nic.go [hunk 5]
```diff
@@ -161,6 +170,15 @@ func resourceScalewayInstancePrivateNICDelete(ctx context.Context, d *schema.Res
 		return diag.FromErr(err)
 	}
 
+	_, err = instanceAPI.WaitForPrivateNIC(&instance.WaitForPrivateNICRequest{
```
Comment (remyleone) [inline scaleway/resource_instance_private_nic.go:173]: Should be a wait for like function. Like the ones recently merged in k8s, function, ...

File: go.sum (hunk 1 above)
Comment (Monitob) [reply go.sum:871]: I just dit but this line don't change.

**Review by remyleone — APPROVED** [review]

**Pull request merged**